python/lsst/afw/coord/refraction.py

```diff
@@ -0,0 +1,284 @@
+                        7.75141E-4*u.Kelvin**3/temperature**3.)
+
+    densityEqn2 = float(waterVaporPressure/cds.mbar)*(1. + 3.7E-4*waterVaporPressure/cds.mbar)
+
+    relativeDensity = float(waterVaporPressure*u.Kelvin/(temperature*cds.mbar))
+    densityFactor = (1 + densityEqn2*densityEqn1)*relativeDensity
+
+    return densityFactor
+
+
+def humidityToPressure(weather):
+    """Simple function that converts humidity and temperature to water vapor pressure.
+
+    This replicates equations 18 & 20 of Stone 1996 "An Accurate Method for Computing Atmospheric Refraction"
+
+    Parameters
+    ----------
+    weather : lsst.afw.coord Weather
+        Class containing the measured temperature, pressure, and humidity
+        at the observatory during an observation
+
+    Returns
+    -------
+    float
+        The water vapor pressure in millibar, calculated from the given humidity and temperature.
+    """
+    if np.isnan(weather.getHumidity()):
+        humidity = lsstWeather.getHumidity()
+    else:
+        humidity = weather.getHumidity()
+    x = np.log(humidity/100.0)
+    temperature = _extractTemperature(weather)
+    temperatureEqn1 = (temperature + 238.3*u.Celsius)*x + 17.2694*temperature
+    temperatureEqn2 = (temperature + 238.3*u.Celsius)*(17.2694 - x) - 17.2694*temperature
+    dewPoint = 238.3*float(temperatureEqn1/temperatureEqn2)
+
+    waterVaporPressure = (4.50874 + 0.341724*dewPoint + 0.0106778*dewPoint**2 + 0.184889E-3*dewPoint**3 +
+                          0.238294E-5*dewPoint**4 + 0.203447E-7*dewPoint**5)*133.32239*u.pascal
+
+    return waterVaporPressure
+
+
+def _extractTemperature(weather, useKelvin=False):
+    """Thin wrapper to return the measured temperature from an observation with astropy units attached.
+
+    Parameters
+    ----------
+    weather : lsst.afw.coord Weather
+        Class containing the measured temperature, pressure, and humidity
+        at the observatory during an observation
+    useKelvin : bool, optional
+        Set to True to return the temperature in Kelvin instead of Celsius
+        This is needed because Astropy can't easily convert between Kelvin and Celsius.
+
+    Returns
+    -------
+    astropy.units
+        The temperature in Celsius, unless `useKelvin` is set.
+    """
+    temperature = weather.getAirTemperature()
+    if np.isnan(temperature):
+        temperature = lsstWeather.getAirTemperature()*u.Celsius
+    else:
+        temperature *= u.Celsius
+    if useKelvin:
+        temperature = temperature.to(u.Kelvin, equivalencies=u.temperature())
+    return temperature
+
+
+def _extractPressure(weather):
+    """Thin wrapper to return the measured pressure from an observation with astropy units attached.
+
+    Parameters
+    ----------
+    weather : lsst.afw.coord Weather
+        Class containing the measured temperature, pressure, and humidity
+        at the observatory during an observation
+
+    Returns
+    -------
+    astropy.units
```

Comment:
Are you sure about the type?

    >>> type(1.2345*astropy.units.pascal)
    <class 'astropy.units.quantity.Quantity'>